lib/local-storage.ts

```diff
@@ -73,12 +73,11 @@ const isStorageSupported = ($window: Window, storageType: StorageType) => {
 	return supported;
 };
 
-interface StorageLike {
-	_init?: () => void;
-	clear(): void;
-	getItem(key: string): string | null;
-	setItem(key: string, data: string): void;
-	removeItem(key: string): void;
+export interface StorageLike {
+	clear(): PromiseLike<void> | void;
+	getItem(key: string): PromiseLike<string | null> | string | null;
+	setItem(key: string, data: string): PromiseLike<void> | void;
+	removeItem(key: string): PromiseLike<void> | void;
```

Comment:
How about making all of them promise based?
I can hear someone saying that in TS 4.0 we will (maybe) be able to define typings per environment, but I would say let's keep this unified.